keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -121,7 +173,13 @@ class SqlDeliveryConfigRepository(
       artifacts.forEach { artifact ->
         jooq.insertInto(DELIVERY_CONFIG_ARTIFACT)
           .set(DELIVERY_CONFIG_ARTIFACT.DELIVERY_CONFIG_UID, uid)
-          .set(DELIVERY_CONFIG_ARTIFACT.ARTIFACT_UID, jooq.select(DELIVERY_ARTIFACT.UID).from(DELIVERY_ARTIFACT).where(DELIVERY_ARTIFACT.NAME.eq(artifact.name)))
+          .set(DELIVERY_CONFIG_ARTIFACT.ARTIFACT_UID, jooq
+            .select(DELIVERY_ARTIFACT.UID)
+            .from(DELIVERY_ARTIFACT)
+            .where(DELIVERY_ARTIFACT.NAME.eq(artifact.name))
+            .and(DELIVERY_ARTIFACT.TYPE.eq(artifact.type.name))
+            .and(DELIVERY_ARTIFACT.DELIVERY_CONFIG_NAME.eq(artifact.deliveryConfigName))
+            .and(DELIVERY_ARTIFACT.REFERENCE.eq(artifact.reference)))
```

Comment:
Good catch! Thanks for adding this! Just one question: I thought not all artifacts had a `reference`?

Reply:
All artifacts have a reference, it's a requirement. It defaults to the name.